src/jose/jws/general/verifier.ts

```diff
@@ -0,0 +1,112 @@
+import type { GeneralJws, Signature } from './types';
+import type { JwkPublic, VerifyFn } from '../../types';
+import type { VerificationMethod } from '../../../did/did-resolver';
+
+import { base64url } from 'multiformats/bases/base64';
+import { DIDResolver } from '../../../did/did-resolver';
+import { verify as verifyEd25519 } from '../../algorithms/ed25519';
+import { verify as verifySecp256k1 } from '../../algorithms/secp256k1';
+
+const verifiers: { [key:string]: VerifyFn } = {
+  'Ed25519'   : verifyEd25519,
+  'secp256k1' : verifySecp256k1
+};
+
+type VerificationResult = {
+  signers: string[];
+};
+
+// TODO: add logic to prevent validating duplicate signatures
+export class GeneralJwsVerifier {
+  jws: GeneralJws;
+
+  constructor(jws: GeneralJws) {
+    this.jws = jws;
+  }
```

Comment:
Style: prefer the other non-class style that you normally use (or class with static methods) and pass in the JWS, it feels strange that I have to instantiate a class to verify a JWS, because there is no real state life-cycle.

Reply:
The motivation behind a class driven design is to support the ability to pass the instance around and easily add signatures. e.g.

```js
// instantiated somewhere
const jwsSigner = GeneralJwsSigner.create(payload);

// signature added elsewhere
await jwsSigner.addSignature({ privateJwk, signatureInput });

// signature added elsewhere
await jwsSigner.addSignature(({ privateJwk, signatureInput }));

const jws = jwsSigner.getJws();
```

```js
// example of adding a signature to an existing general Jws
const jwsSigner = new GeneralJwsSigner(jws);
await jwsSigner.addSignature({ privateJwk, signatureInput });
```

```js
// example for if you have all the signature inputs upfront
const jws = GeneralJwsSigner.create(payload)
  .addSignature({ jwkPrivate, signatureInput })
  .addSignature({ jwkPrivate2, signatureInput2 })
  .getJws();
```

lmk what you think. If you're on board i can add these examples and JSDoc to `signer`

Reply:
Got it, as opposed to passing the `jws` around instead? I guess method chaining is making it look nicer. So, the `verifier` here follows the same pattern as the `signer` mainly for consistency? I hope there is a real need for adding signatures in multiple places, I am not seeing it, bit you can make the call since you obviously thought it about it.

Reply:
It was @csuwildcat actually